Sources/RichEditorView/Resources/editor/rich_editor.js

```diff
@@ -24,6 +24,60 @@ document.addEventListener("selectionchange", function() {
     RE.backuprange();
 });
 
+document.addEventListener('click', function(event) {
+  // This code will be executed whenever a click occurs anywhere inside the document.
+  // You can replace this code with any action you want to perform when a click happens.
+     var items = [];
+
+     if (document.queryCommandState('bold')) {
+         items.push('bold');
+     }
+     if (document.queryCommandState('italic')) {
+         items.push('italic');
+     }
+     if (document.queryCommandState('subscript')) {
+         items.push('subscript');
+     }
+     if (document.queryCommandState('superscript')) {
+         items.push('superscript');
+     }
+     if (document.queryCommandState('strikeThrough')) {
+         items.push('strikeThrough');
+     }
+     if (document.queryCommandState('underline')) {
+         items.push('underline');
+     }
+     if (document.queryCommandState('insertOrderedList')) {
+         items.push('orderedList');
+     }
+     if (document.queryCommandState('insertUnorderedList')) {
+         items.push('unorderedList');
+     }
+     if (document.queryCommandState('justifyCenter')) {
+         items.push('justifyCenter');
+     }
+     if (document.queryCommandState('justifyFull')) {
+         items.push('justifyFull');
+     }
+     if (document.queryCommandState('justifyLeft')) {
+         items.push('justifyLeft');
+     }
+     if (document.queryCommandState('justifyRight')) {
+         items.push('justifyRight');
+     }
+     if (document.queryCommandState('insertHorizontalRule')) {
+         items.push('horizontalRule');
+     }
+     var formatBlock = document.queryCommandValue('formatBlock');
+     if (formatBlock.length > 0) {
+         items.push(formatBlock);
+     }
+
+     const allAppliedFormat = items.toString()
+     RE.callback(allAppliedFormat)
```

Comment:
this isn't ideal. A parsable list would be better or performing the callback one at a time since you're not getting the formats in the order of encapsulation